offload/test/api/omp_host_pinned_memory.c

```diff
@@ -1,5 +1,7 @@
 // RUN: %libomptarget-compile-run-and-check-generic
 
+// UNSUPPORTED: x86_64-pc-linux-gnu-mpi
```

Comment:
I understand why you can't support the tests x86_64 doesn't support, but why are some additional ones unsupported?

Reply:
In this patch, we do not intend to support shared/unified memory, so we have disabled tests for this functionality. We also did not intend to support pinned memory on the host, but I will add a naive implementation in this patch. However, for it to work, we have to wait for the fix in #91134 to be approved.

We disabled the tests `offload/test/api/omp_indirect_call.c` because our plugin is incompatible with the Device Environment features at this time.

When we introduce support for offloading to other remote accelerators, we will add these currently unsupported features.